SConstruct

```diff
@@ -141,6 +142,16 @@ def RecursiveGlob(pattern, dir_name=buildDirectory):
 	matches = [i for i in matches if not '{}main.cpp'.format(os.path.sep) in str(i)]
 	return matches
 
+try:
+    env.Tool('compilation_db')
+    cbd = env.CompilationDatabase()
+    Alias('cbd', cbd)
```

Comment:
Do we need the alias? (And shouldn't it be `cdb` for "_C_ompilation _D_ata_B_ase"?)

Reply:
According to the scons docs (https://scons.org/doc/production/HTML/scons-user/ch27.html) we don't need it but it somehow doesn't work and I have no idea why. oh oops thanks

Reply:
From that link, it appears the compilation db should be made variant-aware as well.

Reply:
I have no idea what that means but I figured it out. `scons compile_commands.json` needs to be used to generate the file since we specify a default target explicitly.

Reply:
I mention being variant-aware since we have various `mode` options and some graphics flags as well, which combine into different compiler arguments. The scons doc linked above mentions one may want different CDBs for this scenario.

Reply:
No we don't need to. The only difference between our different modes are flags which do not affect diagnostics so I don't believe it is necessary.